vendor/bat-native-ads/src/bat/ads/internal/ad_serving/ad_notifications/ad_notification_serving.cc

```diff
@@ -129,24 +129,52 @@ void AdServing::MaybeServeAd() {
         }
 
         BLOG(1, "Served ad notification");
-        ServedAd();
+        ServedAd(ad);
       });
 }
 
 void AdServing::OnAdsPerHourChanged() {
   const int64_t ads_per_hour = settings::GetAdsPerHour();
   BLOG(1, "Maximum ads per hour changed to " << ads_per_hour);
 
-  if (!PlatformHelper::GetInstance()->IsMobile()) {
+  UpdateServingAdsAtRegularIntervals();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+bool AdServing::ShouldServeAdsAtRegularIntervals() {
+  return PlatformHelper::GetInstance()->IsMobile();
+}
+
+void AdServing::UpdateServingAdsAtRegularIntervals() {
+  if (!ShouldServeAdsAtRegularIntervals()) {
     return;
   }
 
-  MaybeServeNextAd();
+  const int64_t ads_per_hour = settings::GetAdsPerHour();
+  if (ads_per_hour == 0) {
+    StopServingAdsAtRegularIntervals();
+    return;
+  }
+
+  MaybeServeAdAtNextInterval();
 }
 
-///////////////////////////////////////////////////////////////////////////////
+void AdServing::RetryServingAdAtNextInterval() {
+  if (!ShouldServeAdsAtRegularIntervals()) {
+    return;
+  }
+
+  const base::TimeDelta delay = base::TimeDelta::FromMinutes(2);
```

Comment:
should we move this to a constant in anonymous namespace?
constexpr base::TimeDelta

Reply:
Fixed